### DISCUSSION TITLE
Why does the String varchar length defaults to 191?

### DISCUSSION THREAD

**Discussion by baptisteArno** [prisma/prisma#17781] [Q&A] [answered]
By default, prisma migrate will convert String to VARCHAR(191).
https://www.prisma.io/docs/concepts/database-connectors/mysql#type-mapping-between-mysql-to-prisma-schema

**Is there any particular reason for this default length?**

I've read that the VARCHAR length [won't affect query performance](https://stackoverflow.com/questions/8295131/best-practices-for-sql-varchar-column-length). Wouldn't it be better to default it to max length to avoid bad surprise in production?

**Comment by janpio** [comment, marked as answer]
This, roughly: https://www.grouparoo.com/blog/varchar-191 We are bound to the "safe conventions" there.

If you want to use a different value, you can supply the native type yourself - and then can also choose other lengths for your varchar fields.

**Reply by baptisteArno** [reply]
Thank you for the clarification 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
